src/main/compatibility-check.ts

```diff
@@ -77,9 +84,12 @@ const checkDatadirVersion = async (): Promise<DatadirCompatibility> => {
   }
 }
 
-export async function checkDatadirCompatibility(t: TFunctionMain): Promise<void> {
+export async function checkDatadirCompatibility(): Promise<DatadirChecked | void> {
```

Comment:
Shouldn't it be more something like `prepareDataDir(rawDatadir: string): Promise<DataDir|void>`?
If the type returned from it contains also a reference to datadir to be used, then it could even be used to put that value back to config.